packaging/systemd/configure-ovs.sh

```diff
@@ -477,284 +145,25 @@ get_bridge_physical_interface() {
-     [ -f "${extra_bridge_file}" ] &&
-     [ "$(cat "${iface_default_hint_file}")" == "$(cat "${extra_bridge_file}")" ]; then
-    echo "${iface_default_hint_file} and ${extra_bridge_file} share the same content"
-    echo "Deleting file ${iface_default_hint_file} to choose a different interface"
-    rm -f "${iface_default_hint_file}"
-    rm -f /run/configure-ovs-boot-done
-  fi
-  # on every boot we rollback and generate the configuration again, to take
-  # in any changes that have possibly been applied in the standard
-  # configuration sources
-  if [ ! -f /run/configure-ovs-boot-done ]; then
-    echo "Running on boot, restoring previous configuration before proceeding..."
-    rollback_nm
-    print_state
-  fi
-  touch /run/configure-ovs-boot-done
-  iface=$(get_default_interface "${iface_default_hint_file}" "$extra_bridge_file")
-  if [ "$iface" != "br-ex" ]; then
-    # Default gateway is not br-ex.
-    # Some deployments use a temporary solution where br-ex is moved out from the default gateway interface
-    # and bound to a different nic (https://github.com/trozet/openshift-ovn-migration).
-    # This is now supported through an extra bridge if requested. If that is the case, we rollback.
-    # We also rollback if it looks like we need to configure things, just in case there are any leftovers
-    # from previous attempts.
-    if [ -f "$extra_bridge_file" ] || [ -z "$(nmcli connection show --active br-ex 2> /dev/null)" ]; then
-      echo "Bridge br-ex is not active, restoring previous configuration before proceeding..."
-      rollback_nm
-      print_state
-    fi
-  fi
-  convert_to_bridge "$iface" "br-ex" "phys0" "${BRIDGE_METRIC}"
-  # Remove the second bridge
-  if nmcli connection show br-ex1 &> /dev/null || nmcli connection show ovs-if-phys1 &> /dev/null; then
-    update_nm_conn_files br-ex1 phys1
-    rm_nm_conn_files
+for file in "${MANAGED_NM_CONN_FILES[@]}"; do
+  if [ -f "$file" ]; then
+    nm_need_rollback=true
   fi
-  # Remove the second bridge hint file
-  if [ -f "$extra_bridge_file" ]; then
-    rm -f "$extra_bridge_file"
-  fi
-  # Remove bridges created by openshift-sdn
-  ovs-vsctl --timeout=30 --if-exists del-br br0
-  # Make sure everything is activated
-  connections=()
-  while IFS= read -r connection; do
-    if [[ $connection == *"$MANAGED_NM_CONN_SUFFIX" ]]; then
-      connections+=("$connection")
-    fi
-  done < <(nmcli -g NAME c)
-  connections+=(ovs-if-phys0 ovs-if-br-ex)
-  activate_nm_connections "${connections[@]}"
-elif [ "$1" == "OpenShiftSDN" ]; then
-  # Revert changes made by /usr/local/bin/configure-ovs.sh during SDN migration.
+done
+
+if [ "${nm_need_rollback}" = "true" ]; then
+  # Remove NM config files create by previous release when doing upgrade
   rollback_nm
-  
-  # Remove bridges created by ovn-kubernetes
-  ovs-vsctl --timeout=30 --if-exists del-br br-int -- --if-exists del-br br-local
 fi
+
+ovs-vsctl --timeout=15 --may-exist add-br br-ex
```

Comment:
I am curious to know if there is a specific reason this command is duped here and 2 lines below

Reply:
Good catch. It shall be removed.